src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -67,6 +45,32 @@
     private final PostConsentRequest postConsentRequest;
     private final PatientServiceClient patientServiceClient;
 
+    @Scheduled(cron = "${cron.expression}")
+    @Async
+    public void processExpiredConsents() {
+        logger.info("Processing consent expiries asynchronously. "
+                + Thread.currentThread().getName());
+        List<ConsentExpiry> consentExpiries = consentArtefactRepository.getConsentArtefacts(GRANTED).collectList().block();
+        if (consentExpiries != null) {
+            consentExpiries.forEach(consentExpiry -> {
+                if (isConsentExpired(consentExpiry.getConsentExpiryDate())) {
+                    ConsentRepresentation consentRepresentation = getConsentRepresentation(consentExpiry.getConsentId(), consentExpiry.getPatientId()).block();
+                    if (consentRepresentation != null) {
+                        ConsentRequestDetail consentRequestDetail = consentRequestRepository.requestOf(
+                                consentRepresentation.getConsentRequestId(),
+                                GRANTED.toString(),
+                                consentRepresentation.getConsentDetail().getPatient().getId()).block();
+                        updateAndBroadcastConsentExpiry(
+                                consentExpiry.getConsentId(),
+                                consentExpiry.getPatientId(),
+                                consentRepresentation,
```

Comment:
It would be better if you move it to separate class.

What would happen if we there are significant number of consents?

Reply:
It is running in its own thread, this method is made async.

Reply:
Own thread is fine, but we are returning is list of ConsentExpiry, what happens if it's more than ~1 MB.

What you have done may be fine now, if you can't address the issue as part of this story, create a tech debt at least.

Reply:
It is running in a separate thread. This method is made async.

Reply:
Moved it to a new class called ConsentScheduler